### PR TITLE
Jet Finite Differences Test, WIP primitives and symbolic zeros

### PR DESCRIPTION
**Pull request by jessebett**
I've corrected the test function to use higher order finite differences method. This will now numerically test the jets for terms other than `[1,0,0,...]` which was the case for the `jvp` based tests previously. I'm unsure of how to write a `jvp` based test that does the equivalent.

I am writing some primitives but am stuck since our previous `make_derivs` required using numpy, e.g. `np.sum` and `np.array`. Since these are now going directly into `lax.py` we cannot import these functions. What is the correct approach here?

The symbolic zeros, `ZeroTerm` and `ZeroSeries` are broken. `ZeroTerm` needs `*` with float defined. `ZeroSeries` needs to support iteration. These are two bugs I've run into so far.

The `total_zhat_jet` test will be to determine how we can use `jet` to give the "total derivative of `f(z(t),t)` wrt t" This uses a known `z(t)` with closed form solution `sol(t)` to test by simply nesting `grad` on the closure with just `t` input. To test we will look at `jet` on the full `f(z_eval,t_eval)`. However, due to the above bugs I am unable to test the simple function `4 * z / t`

### CODE REVIEW

File: tests/total_zhat_jet.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# True total derivatives
+dz = grad(sol)(t_eval)
+ddz = grad(grad(sol))(t_eval)
+dddz = grad(grad(grad(sol)))(t_eval)
+
+# Jet total derivatives
+z_terms = (np.ones_like(z_eval),np.zeros_like(z_eval))
+t_terms = (np.ones_like(t_eval),np.zeros_like(t_eval))
+
+jet(f,(z_eval,t_eval),np.array([z_terms,t_terms]))
```
Comment (jessebett) [inline tests/total_zhat_jet.py]: I can explain this on call, but essentially `sol` provides the closed form solution for `sol(t) = f(z(t),t)` such that `grad` gives the "total derivative" wrt t.

However, we need this total derivative without directly having access to `z(t)`, instead we have access to `z_eval` 

This test will eventually help us determine how to use `jet` on `z_eval` to get the "total derivative" wrt z of `f(z_eval, t_eval)` 

However, this is stopped currently by bugs in my implementation of `mul` and the symbolic zeros.

**Review by jessebett — COMMENTED** [review]
Explaining my work-arounds.

**Comment by duvenaud** [comment]
I think this PR can be deleted now that we have jet in master.

**Comment by mattjj** [comment]
@duvenaud thanks!

**Pull request closed without merging**